sympy/ntheory/factor_.py

```diff
@@ -2468,60 +2442,37 @@ def is_perfect(n):
     .. [2] https://en.wikipedia.org/wiki/Perfect_number
 
     """
+    n = as_int(abs(n))
```

Comment:
Use of `abs` is going to allow this to give an answer when the assumption system would say there should be none (since negatives are not prime).

Reply:
This would also be a change in behavior from master where `is_perfect(-28) -> False`